### DISCUSSION TITLE
如何方便有效地使用自己之前自定义的命令？

### DISCUSSION THREAD

**Discussion by ymzang** [obster-y/XJTU-thesis#80] [Q&A] [answered]
作者您好！
由于我本人研究的课题比较一致，因此在之前的科研中，已经有了一个自己常用的自定义命令集合（文件为.tex，其中绝大部分命令是`\newcommand`）。
我前一段时间在写博士论文的时候，为了使用自己之前的LaTeX代码（尤其是公式），就将自定义命令集合中的内容添加到了`XJTU-thesis.cls`文件中，但这样也就把`.cls`文件和自定义命令耦合了。
不知道能不能实现采用`\input`或者其他命令，将`.cls`文件和自定义命令`.tex`文件解耦？或者作者有没有其他比较推荐的方案？

**Comment by obster-y** [comment, marked as answer]
可以实现

在 `\begin{document}` 之前，使用 `\input{relative-path-to-your-.tex-file}` 导入你的文件即可

并且如果之后出现了问题，需要增加考虑是否你自定义的命令与模板定义的冲突（虽然可能性极低）

**Comment by ymzang** [comment]
好的，谢谢。
确实有可能会出现冲突的情况，感谢提醒！
